packages/loaders/src/Skeleton.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, { keyframes } from 'styled-components';
+import { transparentize } from 'polished';
+import { retrieveTheme } from '@zendeskgarden/react-theming';
+import {
+  zdColorGrey800,
+  zdColorGrey100,
+  zdColorKale800,
+  zdColorWhite,
+  zdSpacingXxs
+} from '@zendeskgarden/css-variables';
+
+const COMPONENT_ID = 'loaders.skeleton';
+
+const fadeInAnimation = props => keyframes`
+  0% {
+    opacity: 0;
+  }
+
+  60% {
+    opacity: 0;
+  }
+
+  100% {
+    opacity: ${props.dark ? '0.15' : '0.1'};
+  }
+`;
+
+const skeletonAnimation = keyframes`
+  0% {
+    left: -1200px;
+  }
+
+  100% {
+    left: 100%;
+  }
+`;
+
+const StyledSkeleton = styled.div.attrs({
+  'data-garden-id': COMPONENT_ID,
+  'data-garden-version': PACKAGE_VERSION
+})`
+  display: inline-block;
+  position: relative;
+
+  animation: ${fadeInAnimation} 700ms linear; /* stylelint-disable-line */
+  opacity: ${props => (props.dark ? '0.15' : '0.1')};
+
+  border-radius: ${zdSpacingXxs};
+  background-color: ${props => (props.dark ? zdColorWhite : zdColorGrey800)};
+  width: ${props => props.width};
+  height: ${props => props.height};
+  overflow: hidden;
+  line-height: 1;
+
+  &::before {
+    position: absolute;
+    top: 0;
+    left: -1200px;
```

Comment:
Whats's the reasoning for a 1000px pseudo element and positioning it 200px further out? Timings?

Since you're setting this value you don't need to repeat it in the animation:

```js
const fadeInAnimation = props => keyframes`
 100% {
    left: 100%;
  }
`;
```

works the same.

Reply:
demo here https://jsbin.com/powecanuwe/2/edit

Reply:
> Whats's the reasoning for a 1000px pseudo element and positioning it 200px further out? Timings?

It's for timing and the size of the transparent area.

I've updated the keyframe